core/domain/exp_jobs_one_off.py

```diff
@@ -535,3 +536,41 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ImageDataMigrationJob(jobs.BaseMapReduceOneOffJobManager):
+    """ One-off job for migrating the images in the exploration
+        from the GAE to GCS
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileSnapshotContentModel]
+
+    @staticmethod
+    def map(file_snapshot_content_model):
+        instance_id = file_snapshot_content_model.get_unversioned_instance_id()
+        filetype = instance_id[instance_id.rfind(".") + 1:]
+
+        if filetype in ALLOWED_IMAGE_EXTENSIONS:
+            filename = instance_id[instance_id.rfind("/") + 1:]
+            exploration_id = (instance_id[1:])[:(instance_id[1:]).find("/")]
+            filepath = (instance_id[1:])[(instance_id[1:]).find("/") + 1:]
+            file_model = file_models.FileModel.get_model(
+                exploration_id, filepath, False)
+
+            if file_model and file_model.deleted:
```

Comment:
Btw, I'm not actually sure whether there are any deleted files. I think you could do something like:

    if file_model.deleted:
        yield ('Error: found deleted file', file_model.id)

just to verify. And then if we find such files we can figure out what to do about them. For now I'm not sure yet, but let's first figure out whether these exist in the first place.

Reply:
Done!